### PR TITLE
Adds option to specify an alternate baseUrl (e.g. CloudFront)

### PR DESCRIPTION
**Pull request by flavordaaave**
The option is optionally, based on the requirement from https://github.com/ParsePlatform/parse-server/issues/792

### CODE REVIEW

**Comment by drew-gross** [comment]
@skinp can you take a look at this? It seems to make sense to me but I don't have much experience with this kind of thing.

File: README.md [hunk 1]
```diff
@@ -86,8 +88,9 @@ var s3Options = {
    	// optional:
    	"region": 'us-east-1', // default value
    	"bucketPrefix": '', // default value
-   	"directAccess": false // default value
-} 
+   	"directAccess": false, // default value
+    "baseUrl": null // default value
```
Comment (skinp) [inline README.md]: nit indent

File: README.md [hunk 2]
```diff
@@ -59,12 +60,13 @@ And update your config / options
...
-								'secretKey', 
+var s3Adapter = new S3Adapter('accessKey',
+								'secretKey',
 								'bucket' , {
-								    region: 'us-east-1'
+								  region: 'us-east-1'
 									bucketPrefix: '',
-									directAccess: false
+									directAccess: false,
+                  baseUrl: 'http://images.example.com'
```
Comment (skinp) [inline README.md]: looks like indent got messed up

**Comment by skinp** [comment]
@drew-gross yes it makes sense to me too and looks good on the S3 side. Was planning on adding exactly this at some point so +1 to @flavordaaave for doing that. 

I'm not sure why code coverage dropped though, maybe we want to look into it...

**Comment by drew-gross** [comment]
Covecov seems to actually just be a random number generator sometimes. I will merge when the indentations issues are sorted out.

**Comment by flavordaaave** [comment]
@drew-gross indentions have been fixed.

**Comment by flovilmart** [comment]
@skinp seems that the S3 env variables are not properly exported with that test. @flavordaaave can you try rebase on the latest master and push?

**Comment by flavordaaave** [comment]
@drew-gross will this pull request be automatically available in the npm repository or is there anything I can do to assist?

**Comment by drew-gross** [comment]
@flovilmart has been handling releases for this repo. Can you do a release?

**Comment by flovilmart** [comment]
I'll Just need to push a new tag

**Comment by flovilmart** [comment]
sorry for the late response. it's pushed as 1.0.2

**Pull request merged**